wpilibcExamples/src/main/cpp/examples/AprilTagsDetection/cpp/Robot.cpp

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include <cstdio>
+#include <span>
+#include <sstream>
+#include <string>
+#include <thread>
+
+#include <cameraserver/CameraServer.h>
+#include <frc/TimedRobot.h>
+#include <frc/apriltag/AprilTagDetection.h>
+#include <frc/apriltag/AprilTagDetector.h>
+#include <frc/apriltag/AprilTagPoseEstimator.h>
+#include <frc/geometry/Transform3d.h>
+#include <frc/smartdashboard/SmartDashboard.h>
+#include <opencv2/core/core.hpp>
+#include <opencv2/core/types.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <units/angle.h>
+#include <units/length.h>
+
+/**
+ * This is a demo program showing the detection of AprilTags.
+ * The image is acquired from the USB camera, then any detected AprilTags
+ * are marked up on the image and sent to the dashboard.
+ */
+class Robot : public frc::TimedRobot {
+#if defined(__linux__) || defined(_WIN32)
```

Comment:
Why are there preprocessor macros here?

Reply:
For the same reasons that they are in the IntermediateVision example; I don't know what those reason are.

Because of the threading, both those examples only run on Win32 or Linux (including roboRIO).

Reply:
macOS supports threading as well. If I had to guess, it's related to OpenCV.

Reply:
could be. This example is adapted from the IntermediateVision example, which has it. I have no way of testing on MacOS. @PeterJohnson  was the last one to touch those preprocessor directives, and may know if MacOS was tested and failed, or was omitted out of an abundance of caution.

As the [this patch file](https://github.com/wpilibsuite/allwpilib/files/10406214/IntermediateVision-to-AprilTagsDetection.patch) shows, this is IntermediateVision with a little bit replaced with the AprilTags detection logic. Any issues with threading are present in both places.

If we want to fix this, we should probably do so in both places (assuming it's fixable), but I think getting this into the hands of team that want it is better than making it perfect. People are prototyping...

Reply:
It's probably because usb cameras didn't work on macOS until this season. I think(?) the macros can be removed now (and on the other vision examples)

Reply:
> Why are there preprocessor macros here?

MacOS still does not work.

Ref: https://www.chiefdelphi.com/t/is-anyone-doing-simulation-on-macos/422804/4